crates/router/src/core/user.rs

```diff
@@ -2560,25 +2565,38 @@ pub async fn create_user_authentication_method(
     }
 
     let now = common_utils::date_time::now();
+    let auth_type = (&req.auth_method).foreign_into();
     state
         .store
         .insert_user_authentication_method(UserAuthenticationMethodNew {
-            id,
-            auth_id,
-            owner_id: req.owner_id,
+            id: id.clone(),
+            auth_id: auth_id.clone(),
+            owner_id: req.owner_id.clone(),
             owner_type: req.owner_type,
-            auth_type: (&req.auth_method).foreign_into(),
+            auth_type: auth_type,
             private_config,
             public_config,
             allow_signup: req.allow_signup,
             created_at: now,
             last_modified_at: now,
-            email_domain,
+            email_domain: email_domain.clone(),
+            email_domain: email_domain.clone(),
         })
         .await
         .to_duplicate_response(UserErrors::UserAuthMethodAlreadyExists)?;
 
-    Ok(ApplicationResponse::StatusOk)
+    let response:user_api::CreateUserAuthenticationMethodResponse =
+        user_api::CreateUserAuthenticationMethodResponse {
+            id: id.clone(),
+            auth_id: auth_id.clone(),
+            owner_id: req.owner_id,
+            owner_type: req.owner_type,
+            auth_type:auth_type,
+            email_domain: Some(email_domain.clone()),
+            allow_signup: req.allow_signup,
+        };
+
+    Ok(ApplicationResponse::Json(response))
```

Comment:
```suggestion
Ok(ApplicationResponse::Json(
    user_api::CreateUserAuthenticationMethodResponse {
        id,
        auth_id,
        owner_id: req.owner_id,
        owner_type: req.owner_type,
        auth_type,
        email_domain: Some(email_domain),
        allow_signup: req.allow_signup,
    },
))
```
Please check if it compiles, we can omit clones at last and return response directly.